### PR TITLE
Improved reliability of Speed Index calculation

### LINKED ISSUE #21: SpeedIndex reported as 0 on sites with <=3 frames
We're seeing some cases where speedindex is reporting 0. Typically happens with sites like airhorner.com and example.com which are very simple and may have only a few frames..

We've seen these sorts of results:

```
{
  'frames': [...],
  'first': 352,
  'complete': 352,
  'duration': 3417,
  'speedIndex': 352
}
```

```
{
  'frames': [...],
  'first': 310,
  'complete': 310,
  'duration': 3366,
  'speedIndex': 0
}
```

cc @brendankenny

### PR DESCRIPTION
**Pull request by paulirish**
Apologies for the size of the PR. I can attempt to break it apart, though it may not be neccessary.

Summary of the changes:

1. **Fix various definitions of `beginning`/time origin**. In a few places, we used the first frame's timestamp as if that were the time origin. Now the first event's timestamp is used as the time origin all the way through. (A later PR [will change this](https://github.com/pmdartus/speedline/issues/27))
1. **Added test cases for traces with 1,2,3.** Added tests for them. The assertions confirm the end speedIndxes; I can imagine we assert something less end-to-end, like progress of the 1st and 2nd frames. wdyt?
1. **Speed Index definition is updated.** It's now, as Parvez [described here](https://github.com/pmdartus/speedline/issues/28#issuecomment-244127192), `SI = firstPaint + sum(fP to VC){1-VC%}`. This addresses the one-frame case completely, and improves accuracy in other places.
1. **Inject a fake white frame**. It has to be the same size as the rest, but we do this for accuracy, as `calculatePerceptualProgress` finds our `minPreceptualProgress` which must be for a blank/default baseline rather than a frame with content.
1. **Removed the dependency on babel**. Aside from `import` statements, there was nothing left that isn't supported in node 5, so the improvement to dev iteration speed seemed worth it.

Fixes #21, #28

------------

Still to do: #27 - allow for a custom time origin (or find navStart ourselves)

Also I reported a bug in babar which leads to bad plots: https://github.com/stephan83/babar/issues/2  Always doublecheck the individual frame progress %s.

### CODE REVIEW

**Comment by pmdartus** [comment]
Nice job. LGTM!

File: package.json [hunk 1]
```diff
@@ -36,19 +34,12 @@
   },
   "devDependencies": {
     "ava": "^0.14.0",
-    "babel-cli": "^6.6.5",
```
Comment (pmdartus) [inline package.json:39]: 👍

File: src/index.js [hunk 2]
```diff
@@ -1,40 +1,23 @@
...
-
-	const {speedIndex, perceptualSpeedIndex} = calculateSpeedIndexes(frames);
+function calculateValues(frames, data) {
+	const indexes = speedIndex.calculateSpeedIndexes(frames, data);
+	const duration = Math.floor(data.endTs - data.startTs);
+	const first = Math.floor(indexes.firstPaintTs - data.startTs);
+	const complete = Math.floor(indexes.visuallyCompleteTs - data.startTs);
 
 	return {
+		beginning: data.startTs,
```
Comment (pmdartus) [inline src/index.js:13]: We should document those fields in the README doc.

Comment (paulirish) [reply src/index.js:13]: Yah good call.

Comment (pmdartus) [reply src/index.js:13]: Merging this for now. I will open a new PR with the doc update.

**Pull request merged**